pkgs/development/python-modules/htpy/default.nix

```
@@ -0,0 +1,59 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  pythonOlder,
+  hatchling,
+  hatch-vcs,
+  markupsafe,
+  pytestCheckHook,
+  django,
+  flask,
+  starlette,
+  httpx,
+  black,
+  ruff,
+  markdown,
+}:
+
+buildPythonPackage rec {
+  pname = "htpy";
+  version = "25.10.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.10";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-y9L3diWTTclpG+rbnpnBkLExBix/6Y/gkBeLgbT39gk=";
+  };
+
+  build-system = [
+    hatchling
+    hatch-vcs
+  ];
+
+  dependencies = [
+    markupsafe
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    django
+    flask
+    starlette
+    httpx
+    black
+    ruff
+    markdown
+  ];
+
+  pythonImportsCheck = [ "htpy" ];
+
+  meta = with lib; {
+    description = "Generate HTML in Python without a template language";
+    license = licenses.mit;
+    homepage = "https://htpy.dev";
+    maintainers = with maintainers; [ dunderrrrrr ];
+  };
```

```suggestion
  meta = {
    description = "Generate HTML in Python without a template language";
    license = lib.licenses.mit;
    homepage = "https://htpy.dev";
    maintainers = with lib.maintainers; [ dunderrrrrr ];
  };
```
`with` before attrsets (especially `with lib;`, _especially_ file-wide ones, _**especially**_ multiple ones) has been abused to death in the past and is now frowned upon[^1]. Just prepend `lib.` where necessary

[^1]: `with` before small scopes, like lists, is fine